code/_onclick/click.dm

```diff
@@ -412,10 +412,10 @@
 	A.AltClick(src)
 
 /atom/proc/AltClick(mob/user)
-	if(!can_interact(user))
-		return FALSE
 	if(SEND_SIGNAL(src, COMSIG_CLICK_ALT, user) & COMPONENT_CANCEL_CLICK_ALT)
 		return
+	if(!can_interact(user))
+		return FALSE
```

Comment:
I really really really do not like this.
Mostly because can_interact always assuming you want powered checks is silly, and this smells like the wrong fix

But also because this means that we send these signals if someone alt clicks on anything, ever.
Even if it's out of range.

Forcing every implementation to check distance seems fraught, and not something I really trust feature coders to do

Reply:
i intended to initiatally just do a bypass in the machines interact proc to allow interaction if its just for rotation rather than checking for power since rotation doesnt need power. but i couldnt figure out how that would even be possible, if(signal == rotation) or something didnt seem to be something that would work, the easiest fix was just to send the signal that alt is being pressed before trying to run a power check, since realistically any proc using alt click that needs to check for power should already inherently be calling an can_interact() check